theano/gpuarray/dnn.py

```diff
@@ -1282,6 +1282,11 @@ def dnn_pool(img, ws, stride=(1, 1), mode='max', pad=(0, 0)):
 
     """
     img = gpu_contiguous(img)
+    if mode == "sum":
+        ret = GpuDnnPool(mode="average_inc_pad")(img, ws, stride, pad)
+        context_name = ret.type.context_name
+        window_elem = numpy.prod(ws).astype(ret.dtype)
```

Comment:
Yes, but numpy.prod work on tuple of python int and Theano variables. I
have checked that before doing the commit.

On Fri, Sep 2, 2016 at 4:31 PM, Pascal Lamblin <notifications@github.com>
wrote:

> In theano/gpuarray/dnn.py
> <https://github.com/Theano/Theano/pull/4932#discussion_r77407509>:
>
> > @@ -1282,6 +1282,11 @@ def dnn_pool(img, ws, stride=(1, 1), mode='max', pad=(0, 0)):
> >
> >      """
> >      img = gpu_contiguous(img)
> > +    if mode == "sum":
> > +        ret = GpuDnnPool(mode="average_inc_pad")(img, ws, stride, pad)
> > +        context_name = ret.type.context_name
> > +        window_elem = numpy.prod(ws).astype(ret.dtype)
>
> ws can be symbolic now, right? It may be the case when the test fails on
> Jenkins.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/Theano/Theano/pull/4932/files/61a1c69f024c31a69eaeb993e7ca3ceb6b9e1057#r77407509>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AALC-8rqhbxiEunTW_H3hU3Jsp6HO9yBks5qmIemgaJpZM4Jz8eJ>
> .
>